streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -167,42 +179,44 @@ public Sensor addThroughputSensor(final String scopeName,
-        addThroughputMetrics(scopeName, sensor, opName, tags);
-    }
-
-    private void addThroughputMetrics(final String scopeName,
-                                      final Sensor sensor,
-                                      final String opName,
-                                      final Map<String, String> tags) {
-        final MetricName rateMetricName = metrics.metricName(opName + "-rate", groupNameFromScope(scopeName),
-            "The average number of occurrence of " + opName + " operation per second.", tags);
-        final MetricName totalMetricName = metrics.metricName(opName + "-total", groupNameFromScope(scopeName),
-            "The total number of occurrence of " + opName + " operations.", tags);
-        if (!metrics.metrics().containsKey(rateMetricName) && !metrics.metrics().containsKey(totalMetricName)) {
-            sensor.add(new Meter(new Count(), rateMetricName, totalMetricName));
-        } else {
-            log.trace("Trying to add metric twice: {} {}", rateMetricName, totalMetricName);
-        }
+    private void addLatencyMetrics(final String scopeName, final Sensor sensor, final String opName, final Map<String, String> tags) {
+        sensor.add(
+            metrics.metricName(
+                opName + "-latency-avg",
+                groupNameFromScope(scopeName),
+                "The average latency of " + opName + " operation.", tags),
+            new Avg()
+        );
+        sensor.add(
+            metrics.metricName(
+                opName + "-latency-max",
+                groupNameFromScope(scopeName),
+                "The max latency of " + opName + " operation.",
+                tags
+            ),
+            new Max()
+        );
     }
 
-    /**
-     * Register a metric on the sensor if it isn't already there.
-     *
-     * @param sensor The sensor on which to register the metric
-     * @param name   The name of the metric
-     * @param stat   The metric to track
-     * @throws IllegalArgumentException if the same metric name is already in use elsewhere in the metrics
-     */
-    public void maybeAddMetric(final Sensor sensor, final MetricName name, final MeasurableStat stat) {
-        sensor.add(name, stat);
+    private void addThroughputMetrics(final String scopeName, final Sensor sensor, final String opName, final Map<String, String> tags) {
+        sensor.add(
+            metrics.metricName(
+                opName + "-rate",
+                groupNameFromScope(scopeName),
+                "The average number of occurrence of " + opName + " operation per second.",
+                tags
+            ),
+            new Rate(TimeUnit.SECONDS, new Count())
+        );
+        sensor.add(
+            metrics.metricName(
+                opName + "-total",
+                groupNameFromScope(scopeName),
+                "The total number of occurrence of " + opName + " operations.",
+                tags
+            ),
+            new Count()
```

Comment:
Similar to the metrics in StreamThread, by getting rid of the Meter and flattening these metrics, I realized that the total computation was incorrectly a Total rather than a Count.

Reply:
Not sure I understand this comment: it was indeed defined as a `Count()` before as well right?

Reply:
I see. That makes sense.

I think it was not introducing any issue only because we only call that `sensor.record()` not `sensor.record(n)` so `Count` and `Total` are actually the same: `record()` is the same as `record(1)`, but I agree that it should really be `Count`, to avoid any potential bugs.